oracles/engine.go

```diff
@@ -64,6 +64,14 @@ func NewEngine(
 	return e
 }
 
+// ListensToPubKeys checks if the public keys from provided OracleData are among the keys
+// current OracleSpecs listen to.
+func (e *Engine) ListensToPubKeys(data OracleData) bool {
+	return e.subscriptions.hasAnySubscribers(func(spec OracleSpec) bool {
```

Comment:
I'm a bit late now its merged (sorry), but given `CheckTx` happens concurrently with the block processing I think we're going to need a lock on the subscribers. We could be adding a new subscriber (due to a new market creation) while at the same time an oracle-command is being checked.